src/util.js

```diff
@@ -186,6 +189,12 @@ export function getDemo(name) {
   return demo;
 }
 
+export function demoSupportsLanguage(demo, id) {
+  const language = getLanguage(id);
+  const demo_dir = demos_dir.get_child(demo.name);
+  return demo_dir.get_child(language.default_file).query_exists(null);
+}
```

Comment:
I was also worried about all the I/O, and although it did not seem to slow it down that much, with a hundred more demos it will likely be a problem.

I thought about adding the languages available in each `.json` file but it is something that can be easily forgotten when adding/porting a demo. Doing it in build time is definitely better.